grpc/resolver.go

```diff
@@ -0,0 +1,112 @@
+		parsedAddress, err := parseResolverIPAddress(address)
+		if err != nil {
+			return nil, err
+		}
+		resolverAddrs = append(resolverAddrs, *parsedAddress)
+	}
+	return newStaticResolver(cc, resolverAddrs), nil
+}
+
+// Scheme returns the scheme that `staticBuilder` will be registered for, for
+// example: `static:///`.
+func (sb *staticBuilder) Scheme() string {
+	return "static"
+}
+
+// staticResolver is used to wrap an inner `resolver.ClientConn` and implements
+// the `resolver.Resolver` interface.
+type staticResolver struct {
+	cc resolver.ClientConn
+}
+
+// newStaticResolver takes a `resolver.ClientConn` and a list of
+// `resolver.Addresses`. It updates the state of the `resolver.ClientConn` with
+// the provided addresses and returns a `staticResolver` which wraps the
+// `resolver.ClientConn` and implements the `resolver.Resolver` interface.
+func newStaticResolver(cc resolver.ClientConn, resolverAddrs []resolver.Address) resolver.Resolver {
+	cc.UpdateState(resolver.State{Addresses: resolverAddrs})
+	return &staticResolver{cc: cc}
+}
+
+// ResolveNow is a no-op necessary for `staticResolver` to implement the
+// `resolver.Resolver` interface. This resolver is constructed once by
+// staticBuilder.Build and the state of the inner `resolver.ClientConn` is never
+// updated.
+func (sr *staticResolver) ResolveNow(_ resolver.ResolveNowOptions) {}
+
+// Close is a no-op necessary for `staticResolver` to implement the
+// `resolver.Resolver` interface.
+func (sr *staticResolver) Close() {}
+
+// parseResolverIPAddress takes an IPv4/6 address (ip:port, [ip]:port, or :port)
+// and returns a properly formatted `resolver.Address` object. The `Addr` and
+// `ServerName` fields of the returned `resolver.Address` will both be set to
+// host:port or [host]:port if the host is an IPv6 address.
+func parseResolverIPAddress(addr string) (*resolver.Address, error) {
+	if addr == "" {
+		return nil, errors.New("address is an empty string")
+	}
+	host, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, fmt.Errorf("splitting host and port for address %q: %w", addr, err)
+	}
+	if port == "" {
+		// If the port field is empty the address ends with colon (e.g.
+		// "[::1]:").
+		return nil, fmt.Errorf("address %q missing port after port-separator colon", addr)
+	}
+	if host == "" {
+		// Address only has a port (i.e ipv4-host:port, [ipv6-host]:port,
+		// host-name:port). Keep consistent with net.Dial(); if the host is
+		// empty, as in (e.g. :80), the local system is assumed.
+		host = "127.0.0.1"
+	}
+	if net.ParseIP(host) == nil {
+		// Host is a DNS name or an IPv6 address without brackets.
+		return nil, fmt.Errorf("address %q is not an IP address", addr)
+	}
+	var parsedAddr string
+	if net.ParseIP(host).To4() != nil {
+		// Host is an IPv4 address.
+		parsedAddr = host + ":" + port
+	} else {
+		// Host is an IPv6 address.
+		parsedAddr = "[" + host + "]:" + port
+	}
```

Comment:
Sorry I didn't realize this earlier!
```suggestion
	parsedAddr := net.JoinHostPort(host, port)
```